Subject: Add lua support to ptxdist 1.99: add automake files
From: Erwin Rol <erwin@erwinrol.com>

This patch add lua support to ptxdist 1.99. It has a menu entry under
scripting languages. And a patch that makes it possible to use autoconf
to build.

This part of the patch adds the automake files.

Signed-off-by: Erwin Rol <erwin@erwinrol.com>
Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>

---
 Makefile.am      |    3 +++
 doc/Makefile.am  |    5 +++++
 etc/Makefile.am  |    8 ++++++++
 src/Makefile.am  |   45 +++++++++++++++++++++++++++++++++++++++++++++
 test/Makefile.am |    5 +++++
 5 files changed, 66 insertions(+)

Index: lua-5.1.3/doc/Makefile.am
===================================================================
--- /dev/null
+++ lua-5.1.3/doc/Makefile.am
@@ -0,0 +1,5 @@
+man1_MANS = lua.1 luac.1
+
+EXTRA_DIST = \
+	contents.html  lua.1   luac.html  lua.html     manual.html \
+	logo.gif       luac.1  lua.css    readme.html
Index: lua-5.1.3/etc/Makefile.am
===================================================================
--- /dev/null
+++ lua-5.1.3/etc/Makefile.am
@@ -0,0 +1,8 @@
+include_HEADERS = lua.hpp
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = lua.pc
+
+EXTRA_DIST = \
+	all.c      lua.ico    lua.pc.in  noparser.c  strict.lua \
+	luavs.bat  min.c      README
Index: lua-5.1.3/Makefile.am
===================================================================
--- /dev/null
+++ lua-5.1.3/Makefile.am
@@ -0,0 +1,3 @@
+SUBDIRS = src doc etc test
+
+EXTRA_DIST = autogen.sh COPYRIGHT HISTORY INSTALL README
Index: lua-5.1.3/src/Makefile.am
===================================================================
--- /dev/null
+++ lua-5.1.3/src/Makefile.am
@@ -0,0 +1,45 @@
+AM_CFLAGS = -Wall
+
+include_HEADERS = lua.h lualib.h lauxlib.h
+nodist_include_HEADERS = luaconf.h
+
+lib_LTLIBRARIES = liblua.la
+liblua_la_LDFLAGS = -release 5.1.3
+liblua_la_SOURCES = \
+	lapi.c lcode.c ldebug.c ldo.c ldump.c lfunc.c lgc.c llex.c lmem.c \
+	lobject.c lopcodes.c lparser.c lstate.c lstring.c ltable.c ltm.c  \
+	lundump.c lvm.c lzio.c \
+	lauxlib.c lbaselib.c ldblib.c liolib.c lmathlib.c loslib.c ltablib.c \
+	lstrlib.c loadlib.c linit.c \
+	lapi.h ldebug.h lgc.h lmem.h lparser.h ltable.h lzio.h ldo.h llex.h \
+	lobject.h lstate.h ltm.h lundump.h lcode.h lfunc.h llimits.h lopcodes.h \
+	lstring.h lvm.h
+
+bin_PROGRAMS = lua luac
+
+lua_SOURCES = lua.c
+lua_LDADD = liblua.la @LUA_LIBS@
+lua_DEPENDENCIES = liblua.la
+
+luac_SOURCES = luac.c print.c
+luac_LDADD = liblua.la @LUA_LIBS@
+luac_DEPENDENCIES = liblua.la
+
+# luac must be linked statically because it uses functions that
+# don't get exported to the shared object
+luac_LDFLAGS = -static
+
+EXTRA_DIST = luaconf.h.template.in
+BUILT_SOURCES = luaconf.h
+CLEANFILES = luaconf.h
+
+readline_defs = @READLINE_DEFS@
+
+edit = 	sed \
+	  -e 's,%lua_datadir%,$(datadir),g' \
+	  -e 's,%lua_libdir%,$(libdir),g'
+
+luaconf.h : luaconf.h.template
+	rm -f luaconf.h luaconf.h.tmp
+	$(edit) luaconf.h.template > luaconf.h.tmp
+	mv luaconf.h.tmp luaconf.h
Index: lua-5.1.3/test/Makefile.am
===================================================================
--- /dev/null
+++ lua-5.1.3/test/Makefile.am
@@ -0,0 +1,5 @@
+EXTRA_DIST = \
+	bisect.lua  env.lua         fib.lua      life.lua          printf.lua \
+	sieve.lua   trace-calls.lua cf.lua       factorial.lua     globals.lua \
+	luac.lua    README          sort.lua     trace-globals.lua echo.lua \
+	fibfor.lua  hello.lua       readonly.lua table.lua         xd.lua
